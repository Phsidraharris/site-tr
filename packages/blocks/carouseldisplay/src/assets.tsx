export const slider1 = require("../assets/slider-1.jpg");
export const slider2 = require("../assets/slider-2.jpg");
export const slider3 = require("../assets/slider-3.jpg");
export const backArrow = require("../assets/image_.png");
export const location = require("../assets/location.png");
export const locationCarousel = require("../assets/locationcarousel.png");
export const share = require("../assets/share.png");
export const play = require("../assets/play.png");
export const plus = require("../assets/plus.png");
export const video4 = require("../assets/videonew.mp4");
export const video1 = require("../assets/videotwo.mp4");

