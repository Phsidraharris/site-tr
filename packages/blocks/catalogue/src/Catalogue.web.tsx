// Customizable Area Start
import React from "react";

import {
  Box,
  Button,
  ButtonGroup,
  Grid,
  Typography,
  styled,
} from "@material-ui/core";
import {
  location, bedroom, bathroom, ownership, reception, parking, Conservatory, Balcony, Pool, Exercise,
  basement, garden, wine, solar, Media, officehome, locationImage, company, phone, share, floor, epc
}
  from "./assets";

import CatalogueController, { Props } from "./CatalogueController";
import Footer from "../../landingpage/src/Footer.web";
import ThreeImgCarouselDisplay from "../../carouseldisplay/src/ThreeImgCarouselDisplay.web";

export default class Catalogue extends CatalogueController {
  constructor(props: Props) {
    super(props);
  }

  render() {
    const { selectedButton } = this.state
    return (
      <>
        <Box>
          <ThreeImgCarouselDisplay navigation={this.props.navigation} id={""}/>
        </Box>
        <Box style={webStyle.content}>
          <Grid container spacing={5} style={{ background: "#FFFFFF", marginTop: "10px" }}>
            <Grid item xs={12} sm={6} style={webStyle.priceBox}>
              <div style={{ textAlign: "center" }}>
                <Typography style={webStyle.priceTextName as React.CSSProperties}>Guide price</Typography>
                <Typography style={webStyle.priceText as React.CSSProperties}>£402,999</Typography>
              </div>
            </Grid>
            <Grid item xs={12} sm={6} style={webStyle.sqFeat}>
              <Typography style={webStyle.priceText as React.CSSProperties}>1047.00 sq.ft.</Typography>
            </Grid>
          </Grid>
          <Grid container style={{ marginTop: "65px" }}>
            <Grid item xs={12} sm={12} md={8} >
              <Box style={webStyle.newBuild as React.CSSProperties}>New Build</Box>
              <Typography style={webStyle.name as React.CSSProperties}>
                Beautiful Apartment
              </Typography>
              <Box style={{ marginTop: "10px", display: "flex" }}>
                <img src={location} style={{ width: "20px", height: "24px" }} />
                <Typography style={webStyle.location as React.CSSProperties}>Shree Garden 1132, UK, 756856</Typography>
              </Box>
              <Typography style={webStyle.propertyType as React.CSSProperties}>Farmhouse</Typography>
              <Box style={{ display: "flex", marginTop: "30px" }}>
                <Box style={webStyle.bedBath as React.CSSProperties}>
                  <img src={bedroom} />
                  <Typography style={webStyle.bedBathText as React.CSSProperties}>4 Bedrooms</Typography>
                </Box>
                <Box style={webStyle.bedBath as React.CSSProperties}>
                  <img src={bathroom} />
                  <Typography style={webStyle.bedBathText as React.CSSProperties}>4 Bathrooms</Typography>
                </Box>
                <Box style={webStyle.bedBath as React.CSSProperties}>
                  <img src={ownership} />
                  <Typography style={webStyle.bedBathText as React.CSSProperties}>Shared Ownership</Typography>
                </Box>
                <Box style={webStyle.bedBath as React.CSSProperties}>
                  <img src={reception} />
                  <Typography style={webStyle.bedBathText as React.CSSProperties}>2 Reception Rooms</Typography>
                </Box>
                <Box style={webStyle.bedBath as React.CSSProperties}>
                  <img src={parking} />
                  <Typography style={webStyle.bedBathText as React.CSSProperties}>Parking Available</Typography>
                </Box>
              </Box>
              <Typography style={webStyle.descriptionTitle as React.CSSProperties}>
                Property description
              </Typography>
              <Typography style={webStyle.description as React.CSSProperties}>
                Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged. It was popularized in the 1960s with the release of Letraset sheets containing Lorem Ipsum passages, and more recently with desktop publishing software like Aldus PageMaker including versions of Lorem Ipsum.
                <span style={{ color: "#0097CB" }}> read more..</span>
              </Typography>
              <Box style={webStyle.propDetails}>
                <Typography style={webStyle.detailsTitle as React.CSSProperties}>Feature Details</Typography>
                <Grid container style={webStyle.propertyFeature as React.CSSProperties}>
                  <Grid item xs={12} sm={2} style={webStyle.bedBath as React.CSSProperties}>
                    <img src={Conservatory} />
                    <Typography style={webStyle.bedBathText as React.CSSProperties}>Conservatory</Typography>
                  </Grid>
                  <Grid item xs={12} sm={2} style={webStyle.bedBath as React.CSSProperties}>
                    <img src={basement} />
                    <Typography style={webStyle.bedBathText as React.CSSProperties}>Basement</Typography>
                  </Grid>
                  <Grid item xs={12} sm={2} style={webStyle.bedBath as React.CSSProperties}>
                    <img src={Balcony} />
                    <Typography style={webStyle.bedBathText as React.CSSProperties}>Balcony</Typography>
                  </Grid>
                  <Grid item xs={12} sm={2} style={webStyle.bedBath as React.CSSProperties}>
                    <img src={Pool} />
                    <Typography style={webStyle.bedBathText as React.CSSProperties}>Swimming Pool</Typography>
                  </Grid>
                  <Grid item xs={12} sm={2} style={webStyle.bedBath as React.CSSProperties}>
                    <img src={garden} />
                    <Typography style={webStyle.bedBathText as React.CSSProperties}>Garden</Typography>
                  </Grid>
                  <Grid item xs={12} sm={2} style={webStyle.bedBath as React.CSSProperties}>
                    <img src={Exercise} />
                    <Typography style={webStyle.bedBathText as React.CSSProperties}>Exercise Room</Typography>
                  </Grid>
                  <Grid item xs={12} sm={2} style={webStyle.bedBath as React.CSSProperties}>
                    <img src={Media} />
                    <Typography style={webStyle.bedBathText as React.CSSProperties}>Media Room</Typography>
                  </Grid>
                  <Grid item xs={12} sm={2} style={webStyle.bedBath as React.CSSProperties}>
                    <img src={officehome} />
                    <Typography style={webStyle.bedBathText as React.CSSProperties}>Home Office</Typography>
                  </Grid>
                  <Grid item xs={12} sm={2} style={webStyle.bedBath as React.CSSProperties}>
                    <img src={wine} />
                    <Typography style={webStyle.bedBathText as React.CSSProperties}>Wine Cellar</Typography>
                  </Grid>
                  <Grid item xs={12} sm={2} style={webStyle.bedBath as React.CSSProperties}>
                    <img src={solar} />
                    <Typography style={webStyle.bedBathText as React.CSSProperties}>Solar Panels</Typography>
                  </Grid>
                </Grid>
              </Box>
              <Box style={webStyle.propDetails}>
                <Typography style={webStyle.detailsTitle as React.CSSProperties}>Energy Performance Certificate & Floor Plan</Typography>
                <Box style={{ padding: "20px" }}>
                  <ButtonGroup>
                    <Button
                      data-test-id="button 1"
                      style={{
                        color: selectedButton === 0 ? '#0097CB' : '#787878',
                        borderColor: selectedButton === 0 ? '#0097CB' : '#787878',
                        background: "#FFFFFF",
                        textTransform: "none",
                      }}
                      onClick={() => this.handleButtonClick(0)}
                    >
                      Floor Plan
                    </Button>
                    <Button
                      data-test-id="button 2"
                      style={{
                        color: selectedButton === 1 ? '#0097CB' : '#787878',
                        borderColor: selectedButton === 1 ? '#0097CB' : '#787878',
                        textTransform: "none",
                        background: "#FFFFFF",
                        width: "100px"
                      }}
                      onClick={() => this.handleButtonClick(1)}
                    >
                      EPC
                    </Button>
                  </ButtonGroup>
                  <Box>
                    {selectedButton === 0 ? (
                      <div>
                        <div style={{ display: "flex", justifyContent: "space-between", marginTop: "20px" }}>
                          <Typography style={webStyle.smalltext as React.CSSProperties}>Loreum ipsum</Typography>
                          <Typography style={webStyle.secSmall as React.CSSProperties}>£402,999</Typography>
                        </div>
                        <Typography>Square Footage : 799.00 sq.ft.</Typography>
                        <Box style={{ padding: "50px" }}>
                          <img src={floor} style={{ width: "100%" }} />
                        </Box>
                      </div>
                    )
                      : (
                        <div style={{ marginTop: "15px", padding: "50px" }}>
                          <img src={epc} style={{ width: "100%" }} />
                        </div>
                      )
                    }
                  </Box>
                </Box>
              </Box>
              <Box style={webStyle.propDetails}>
                <Typography style={webStyle.detailsTitle as React.CSSProperties}>Location</Typography>
                <Box style={{ padding: "30px" }}>
                  <img src={locationImage} style={{ width: "100%" }} />
                </Box>
              </Box>
            </Grid>

            <Grid item xs={12} sm={6} md={4}>
              <AgentBox>
                <Grid container>
                  <Grid item xs={12} sm={3}>
                    <img src={company} />
                  </Grid>
                  <Grid item xs={12} sm={8}>
                    <Typography style={webStyle.company as React.CSSProperties}>Foxtail Agency</Typography>
                    <Typography style={webStyle.place as React.CSSProperties}>Hammersmith, London, W6 8KL</Typography>
                    <div style={{ display: "flex" }}>
                      <img src={phone} style={{ width: "12px", height: "17px", paddingTop: 3, marginRight: "5px" }} />
                      <Typography style={webStyle.number as React.CSSProperties}>01256 781234</Typography>
                    </div>
                  </Grid>
                  <Grid item xs={12} sm={1}>
                    <Box style={webStyle.shareIcon}>
                      <img src={share} />
                    </Box>
                  </Grid>
                </Grid>
                <Button style={webStyle.enqiry as React.CSSProperties}>Enquiry form</Button>
                <Button style={webStyle.offer as React.CSSProperties}>Make an Offer</Button>
              </AgentBox>
            </Grid>
          </Grid>
        </Box>
        <Footer navigation={undefined} id={""} />
      </>
      // Customizable Area End
    );
  }
}

// Customizable Area Start
const AgentBox = styled(Box)({
  background: "#FFFFFF", maxWidth: "430px", padding: "37px",
  '@media (max-width: 1180px)': {
    padding: "12px"
  },
  '@media (max-width: 976px)': {
    padding: "2px"
  },
  '@media (max-width: 959px)': {
    padding: "35px",
    marginTop: "30px"
  },
})
const webStyle = {
  content: {
    paddingLeft: "50px",
    paddingRight: "50px",
    background: "#FAFAFA"
  },
  priceBox: {
    borderRight: "1px solid #848FAC",
    display: "flex",
    justifyContent: "center",
    alignItem: "center"
  },
  sqFeat: {
    display: "flex",
    justifyContent: "center",
    alignItem: "center"
  },
  newBuild: {
    width: "133px",
    height: "44px",
    fontFamily: "Poppins",
    backgroundColor: "#273567",
    color: "#FFFFFF",
    display: "flex",
    justifyContent: "center",
    alignItems: "center",
    borderRadius: "8px",
    lineHeight: "24px",
    fontSize: "16px",
    fontWeight: 500,
    textTransform: "none",
    textAlign: "center"
  },
  name: {
    color: "#273567",
    fontFamily: "Poppins",
    fontWeight: 500,
    fontSize: "36px",
    lineHeight: "44px",
    marginTop: "25px",
  },
  location: {
    color: "#848FAC",
    fontFamily: "Poppins",
    fontWeight: 400,
    fontSize: "20px",
    lineHeight: "28px",
    marginLeft: "5px",
  },
  descriptionTitle: {
    fontFamily: "Poppins",
    color: "#273567",
    fontWeight: 600,
    fontSize: "20px",
    lineHeight: "28px",
    marginTop: "25px"
  },
  description: {
    maxWidth: "700px",
    color: "#848FAC",
    fontWeight: 400,
    fontSize: "16px",
    lineHeight: "24px",
    fontFamily: "Poppins",
    marginTop: "25px"
  },
  bedBath: {
    marginRight: "20px",
    textAlign: "center",
    marginBottom: "30px",
  },
  propertyType: {
    marginTop: "10px",
    color: "#848FAC",
    fontWeight: 400,
    fontFamily: "Poppins",
    lineHeight: "28px",
    fontSize: "20px",
  },
  bedBathText: {
    color: "#1E1E1E",
    fontFamily: "Poppins",
    fontSize: "16px",
    lineHeight: "21px",
    fontWeight: 400,
    marginTop: "10px"
  },
  priceText: {
    color: "#000000",
    fontSize: "20px",
    lineHeight: "20px",
    fontWeight: 400
  },
  priceTextName: {
    color: "#273567",
    lineHeight: "20px",
    fontSize: "16px",
    fontWeight: 400
  },
  propDetails: {
    boxShadow: "rgba(100, 100, 111, 0.2) 0px 7px 29px 0px",
    marginTop: "50px",
  },
  detailsTitle: {
    color: "#273567",
    fontSize: "20px",
    lineHeight: "20px",
    fontFamily: "Poppins",
    fontWeight: 600,
    borderBottom: "1px solid #848FAC",
    padding: "25px 25px 30px 25px"
  },
  propertyFeature: {
    paddingTop: "35px",
    paddingLeft: "30px"
  },
  company: {
    fontFamily: "Poppins",
    color: "#000000",
    fontSize: "20px",
    lineHeight: "28px",
    fontWeight: 400
  },
  place: {
    fontFamily: "Poppins",
    fontSize: "16px",
    lineHeight: "24px",
    color: "#848FAC",
    fontWeight: 400
  },
  number: {
    color: "#848FAC",
    fontWeight: 400,
    fontFamily: "Poppins",
    fontSize: "16px",
    lineHeight: "24px"
  },
  shareIcon: {
    boxShadow: "rgba(149, 157, 165, 0.2) 0px 8px 24px",
    maxWidth: "56px",
    maxHeight: "40px"
  },
  enqiry: {
    color: "#273567",
    background: "#FFFFFF",
    border: "1px solid #273567",
    width: "100%",
    fontFamily: "Poppins",
    fontSize: "16px",
    lineHeight: "24px",
    fontWeight: 400,
    textTransform: "none",
    borderRadius: "8px",
    marginTop: "20px",
  },
  offer: {
    color: "#FFFFFF",
    background: "#273567",
    width: "100%",
    fontFamily: "Poppins",
    fontSize: "16px",
    lineHeight: "24px",
    fontWeight: 400,
    marginTop: "10px",
    textTransform: "none",
    borderRadius: "8px",
  },
  smalltext: {
    color: "#000000",
    fontFamily: "Poppins",
    fontSize: "16px",
    fontWeight: 400,
    lineHeight: "24px"
  },
  secSmall: {
    fontFamily: "Poppins",
    color: "#000000",
    lineHeight: "24px",
    fontSize: "12px",
    fontWeight: 400,
  }
};
// Customizable Area End
