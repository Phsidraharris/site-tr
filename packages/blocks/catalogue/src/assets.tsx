export const logo = require("../assets/logo.png");
export const location = require("../assets/location.png");
export const bedroom = require("../assets/bedroom.png");
export const bathroom = require("../assets/bathroom.png");
export const ownership = require("../assets/ownership.png");
export const parking = require("../assets/parking.png");
export const reception = require("../assets/reception.png");
export const Conservatory = require("../assets/Conservatory.png");
export const basement = require("../assets/basement.png");
export const Balcony = require("../assets/Balcony.png");
export const Pool = require("../assets/Pool.png");
export const garden = require("../assets/garden.png");
export const Exercise = require("../assets/Exercise.png");
export const Media = require("../assets/Media.png");
export const officehome = require("../assets/officehome.png");
export const wine = require("../assets/wine.png");
export const solar = require("../assets/solar.png");
export const image = require("../assets/image.png");
export const imgicon = require("../assets/imgicon.png");
export const delet = require("../assets/delete.png");
export const circle = require("../assets/circle.png");
export const cloud = require("../assets/cloud.png");
export const radioCheck = require("../assets/radioCheck.png");
export const radiouncheck = require("../assets/radiouncheck.png");
export const arrow = require("../assets/arrow.png");
export const nextarrow = require("../assets/nextarrow.png");
export const locationImage = require("../assets/locationImage.png");
export const share = require("../assets/share.png");
export const phone = require("../assets/phone.png");
export const company = require("../assets/company.png");
export const epc = require("../assets/EPC.png");
export const floor = require("../assets/floor.png");
export const cross = require("../assets/Icon.png");
export const thumb = require("../assets/thumb.png");
export const play = require("../assets/play.png");
export const tick = require("../assets/tick.png");