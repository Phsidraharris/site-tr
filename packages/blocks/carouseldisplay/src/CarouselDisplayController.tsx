import { IBlock } from "../../../framework/src/IBlock";
import { Message } from "../../../framework/src/Message";
import { BlockComponent } from "../../../framework/src/BlockComponent";
import MessageEnum, {
  getName,
} from "../../../framework/src/Messages/MessageEnum";
import { runEngine } from "../../../framework/src/RunEngine";

// Customizable Area Start
import { slider1, slider2, slider3, video4, video1 } from "./assets";
import ReactPlayer from "react-player";
// Customizable Area End

export const configJSON = require("./config");

export interface Props {
  navigation: any;
  id: string;
  // Customizable Area Start
  // Customizable Area End
}

interface S {
  // Customizable Area Start
  index: number;
  imgData: any;
  groupedImgData: any[];
  isPlaying: boolean;
  isFullCarousel: boolean;
  // Customizable Area End
}

interface SS {
  id: any;
  // Customizable Area Start
  // Customizable Area End
}

export default class CarouselDisplayController extends BlockComponent<
  Props,
  S,
  SS
> {
  // Customizable Area Start
  swiper: any;
  public player: ReactPlayer | null = null;
  // Customizable Area End

  constructor(props: Props) {
    super(props);
    this.receive = this.receive.bind(this);

    // Customizable Area Start
    this.subScribedMessages = [
      // Customizable Area Start
      // Customizable Area End
    ];

    this.state = {
      // Customizable Area Start
      isFullCarousel: false,
      isPlaying: false,
      index: 0,
      imgData: [
        { id: 1, src: slider1, type: "img" },
        { id: 2, src: video1, type: "vid" },
        { id: 3, src: slider2, type: "img" },
        { id: 4, src: slider3, type: "img" },
        { id: 5, src: video4, type: "vid" },
      ],
      groupedImgData: [
        {
          scrOne: { id: 0, src: slider1, type: "img" },
          srcTwo: { id: 1, src: video1, type: "vid" },
          srcThree: { id: 2, src: slider2, type: "img" },
        },
        {
          scrOne: { id: 3, src: slider3, type: "img" },
          srcTwo: { id: 4, src: video4, type: "vid" },
        },
      ],
      // Customizable Area End
    };
    runEngine.attachBuildingBlock(this as IBlock, this.subScribedMessages);

    // Customizable Area Start
    // Customizable Area End
  }

  async receive(from: string, message: Message) {
    // Customizable Area Start
    runEngine.debugLog("Message Recived", message);
    // Customizable Area End
  }

  // Customizable Area Start
  async componentDidMount() {
    this.setState({
      index: parseInt(this.props.navigation.getParam("imgId", 0)),
    });
  }
  handleRef = (ref: any) => {
    this.player = ref;
  };
  handleVideoClick = (e: any) => {
    e.preventDefault();
    this.setState({ isPlaying: !this.state.isPlaying });
  };
  playPauseVideo = (e: any) => {
    if (e.key === " " || e.key === "Spacebar") {
      e.preventDefault();
      this.setState({ isPlaying: !this.state.isPlaying });
    }
  };
  handlePause = () => {
    if (this.player) {
      this.player.getInternalPlayer().focus();
      this.setState({
        isPlaying: true,
      });
    }
  };
  getBreakPoints = (itemTwo: any, itemThree: any) => {
    if (itemTwo) {
      if (itemThree) {
        return 12;
      } else return 7;
    } else return 12;
  };
  getThreeBreakpoint = (itemThree: any) => {
    if (itemThree) {
      return 6;
    } else return 12;
  };
  getCount = () => {
    return (
      (this.state.groupedImgData.length - 2) * 3 +
      Object.keys(
        this.state.groupedImgData[this.state.groupedImgData.length - 1]
      ).length -
      this.state.index * 3
    );
  };
  handleGoToSlide = (id: any) => {
    this.props.navigation.navigate("CarouselDisplay", { imgId: id });
  };
  goBackToCatalogue = () => {
    this.props.navigation.navigate("Catalogue");
  };
  handleShareClick = (e: any) => {
    e.stopPropagation();
  };
  handleLocationClick = (e: any) => {
    e.stopPropagation();
  };
  // Customizable Area End
}
