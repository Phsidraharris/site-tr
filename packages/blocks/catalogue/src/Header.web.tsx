import React from "react";
// Customizable Area Start
import {
  Box,
  Button,
  Link,
} from "@material-ui/core";
import {logo} from './assets'
// Customizable Area End

import CatalogueController, { Props } from "./CatalogueController";

export default class Header extends CatalogueController {
  constructor(props: Props) {
    super(props);
    // Customizable Area Start
    // Customizable Area End
  }

  // Customizable Area Start
  // Customizable Area End

  render() {
    const {activeLink}=this.state
    return (
      //Merge Engine DefaultContainer
      <>
      <Box style={webStyles.header}>
            <img src={logo} style={webStyles.logo} alt="logo"/>
            <Box>
                <Button id="hoverableDark" style={webStyles.listProperty as React.CSSProperties}>
                  List my property
                </Button>
                <Button  variant="contained"
                    color="primary"
                    id="hoverableLight"
                         style={webStyles.myHub as React.CSSProperties}>
                     Log in
                </Button>
            </Box>
        </Box>
        <Box style={webStyles.navbarBox}>
            <Link
              href="#"
              data-test-id="testid1"
              onClick={() => this.handleClick("Browse Properties")}
              style={{
                ...webStyles.navLink,
                color: activeLink === "Browse Properties" ? "#273567" : "#848FAC",
                textDecoration: "none",
                borderBottom: activeLink === "Browse Properties" ? "1px solid #0097CB":"none",
              }}
            >
              Browse Properties
            </Link>
            <Link
              href="#"
              data-test-id="testid2"
              onClick={() => this.handleClick("Valuation")}
              style={{
                ...webStyles.navLink,
                color: activeLink === "Valuation" ? "#273567" : "#848FAC",
                textDecoration: "none",
                borderBottom: activeLink === "Valuation" ? "1px solid #0097CB":"none",
              }}
            >
              Valuation
            </Link>
            <Link
              href="#"
              data-test-id="testid3"
              onClick={() => this.handleClick("Market Insights")}
              style={{
                ...webStyles.navLink,
                color: activeLink === "Market Insights" ? "#273567" : "#848FAC",
                textDecoration: "none",
                borderBottom: activeLink === "Market Insights" ? "1px solid #0097CB":"none",
              }}
            >
              Market Insights
            </Link>
            <Link
              href="#"
              data-test-id="testid4"
              onClick={() => this.handleClick("About us")}
              style={{
                ...webStyles.navLink,
                color: activeLink === "About us" ? "#273567" : "#848FAC",
                textDecoration: "none",
                borderBottom: activeLink === "About us" ? "1px solid #0097CB":"none",
              }}
            >
              About us
            </Link>
          </Box>
        </>
      //Merge Engine End DefaultContainer
    );
  }
}

// Customizable Area Start
const webStyles = {
    header:{
        paddingBottom:"20px",
        borderBottom:"1px solid #2735674D",
        paddingLeft:"50px",
        paddingRight:"50px",
        display:"flex",
        justifyContent:"space-between",
        paddingTop:"20px",
      },
      logo:{    
        width:"129px",
        height:"46px"
      },
      listProperty:{
        backgroundColor:"#141D40",
        borderRadius:"8px",
        color:"#FFFFFF",
        lineHeight:"24px",
        marginRight:"8px",
        height:"44px",
        fontFamily:"Poppins",
        fontWeight:"bold",
        padding: "10px 16px 10px 16px",
        textTransform:"none",
      },
      myHub:{
        border:"1px solid #273567",
        backgroundColor:"F4F5F7",
        color:"#273567",
        borderRadius:"8px",
        height:"44px",
        boxShadow:"none",
        fontFamily:"Poppins",
        fontWeight:"bold",
        textTransform:"none",
        padding: "10px 16px 10px 16px"
      },
      navbarBox: {
        paddingLeft: "50px",
        paddingTop: "20px",
        paddingBottom: "20px",
      },
      navLink: {
        color: "#273567",
        marginRight: "20px",
        pointer: "cursor",
        fontFamily: "Poppins"
      },
};
// Customizable Area End
