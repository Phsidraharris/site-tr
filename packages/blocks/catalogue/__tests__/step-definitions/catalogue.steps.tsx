import { defineFeature, loadFeature } from "jest-cucumber";
import { shallow, ShallowWrapper } from "enzyme";

import * as helpers from "../../../../framework/src/Helpers";

import React from "react";
import Catalogue from "../../src/Catalogue";
const navigation = require("react-navigation");

const screenProps = {
  navigation: navigation,
  id: "Catalogue",
};
export class LocalStorageMock {
  store: { [k: string]: string };
  length: number;

  constructor() {
    this.store = {};
    this.length = 0;
  }

  /**
   * @see https://developer.mozilla.org/en-US/docs/Web/API/Storage/key
   * @returns
   */
  key = (idx: number): string => {
    const values = Object.values(this.store);
    return values[idx];
  };

  clear() {
    this.store = {};
  }

  getItem(key: string) {
    return this.store[key] || null;
  }

  setItem(key: string, value: string) {
    this.store[key] = String(value);
  }

  removeItem(key: string) {
    delete this.store[key];
  }
}

const feature = loadFeature("./__tests__/features/catalogue-scenario.feature");

defineFeature(feature, (test) => {
  beforeEach(() => {
    jest.resetModules();
    jest.doMock("react-native", () => ({ Platform: { OS: "web" } }));
    jest.spyOn(helpers, "getOS").mockImplementation(() => "web");
    global.localStorage = new LocalStorageMock()
  });

  test("User navigates to catalogue", ({ given, when, then }) => {
    let exampleBlockA: ShallowWrapper;
    let instance: Catalogue;

    given("I am a User loading catalogue", () => {
      exampleBlockA = shallow(<Catalogue {...screenProps} />);
    });

    when("I navigate to the catalogue", () => {
      instance = exampleBlockA.instance() as Catalogue;
    });

    then("catalogue will load with out errors", () => {
      expect(exampleBlockA).toBeTruthy();
      expect(exampleBlockA).toMatchSnapshot();
    });

    then("I can leave the screen with out errors", () => {
      instance.componentWillUnmount();
      expect(exampleBlockA).toBeTruthy();
      expect(exampleBlockA).toMatchSnapshot();
    });
  });
});
