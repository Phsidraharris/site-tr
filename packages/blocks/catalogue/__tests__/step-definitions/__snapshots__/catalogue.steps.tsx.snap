// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`catalogue User navigates to catalogue 1`] = `
<ScrollViewMock
  keyboardShouldPersistTaps="always"
  style={
    Object {
      "backgroundColor": "#ffffffff",
      "flex": 1,
      "marginLeft": "auto",
      "marginRight": "auto",
      "maxWidth": 650,
      "width": "100%",
    }
  }
>
  <TouchableWithoutFeedback
    onPress={[Function]}
  >
    <View>
      <View
        style={
          Array [
            Object {
              "backgroundColor": "white",
              "flex": 1,
              "justifyContent": "center",
              "width": "100%",
            },
            Object {
              "display": "flex",
              "flexDirection": "row",
            },
          ]
        }
      >
        <FlatList
          columnWrapperStyle={
            Object {
              "flex": 1,
              "justifyContent": "space-between",
            }
          }
          data={Array []}
          disableVirtualization={false}
          extraData={
            Object {
              "activeLink": "Browse Properties",
              "address": "",
              "agentName": "",
              "arrayHolder": Array [],
              "city": "",
              "description": "",
              "epcrating": "",
              "extensiontype": Array [],
              "inputFieldError": Object {
                "City": false,
                "Postcode One": true,
                "Postcode Two": true,
                "Property Description": false,
                "Property Number": true,
                "Property Price": true,
                "Property Title": false,
                "Streen name": false,
                "keyFeature": false,
                "porpstatus": false,
                "propTenure": false,
                "propType": false,
                "selectedBathroom": false,
                "selectedBed": false,
                "selectedImage": false,
              },
              "isBathroomFilled": true,
              "isBoxOpen": false,
              "isCityFilled": true,
              "isDescriptionFilled": true,
              "isImageBoxOpen": false,
              "isKeyFilled": true,
              "isMainImageFilled": true,
              "isNextButtonDisabled": false,
              "isPostcodeOneFilled": true,
              "isPostcodeTwoFilled": true,
              "isPriceModifierFilled": true,
              "isPropNumberFilled": true,
              "isPropPriceFilled": true,
              "isPropTypeFilled": true,
              "isPropertyTitleFilled": true,
              "isSnackbarOpen": false,
              "isStatusFilled": true,
              "isStreetFilled": true,
              "isTenureFilled": true,
              "isaddressfilled": true,
              "isagentfilled": true,
              "isbedFilled": true,
              "isnumberValid": true,
              "keyFeature": Array [],
              "parking": "",
              "phoneNumber": "",
              "porpstatus": "",
              "postcodeOne": "",
              "postcodeTwo": "",
              "propDesInfo": false,
              "propNumber": "",
              "propPrice": "",
              "propTenure": "",
              "propType": "",
              "propertyTitle": "",
              "reception": "",
              "save": false,
              "selectedAgent": "",
              "selectedBathroom": "",
              "selectedBed": "",
              "selectedButton": 0,
              "selectedFloorImage": null,
              "selectedForDeletion": null,
              "selectedImage": null,
              "selectedImageFilename": "",
              "selectedImageSecond": Array [],
              "selectedImageToDelete": Array [],
              "selectedImgeEpc": null,
              "selectedOption": "",
              "selectedPriceModi": "",
              "selectedPropStatus": "Available",
              "streetname": "",
              "token": "",
              "tourlink": "",
              "userLink": "",
            }
          }
          horizontal={false}
          initialNumToRender={10}
          keyExtractor={[Function]}
          maxToRenderPerBatch={10}
          numColumns={2}
          onEndReachedThreshold={2}
          removeClippedSubviews={false}
          renderItem={[Function]}
          scrollEventThrottle={50}
          updateCellsBatchingPeriod={50}
          windowSize={21}
        />
      </View>
    </View>
  </TouchableWithoutFeedback>
</ScrollViewMock>
`;

exports[`catalogue User navigates to catalogue 2`] = `
<ScrollViewMock
  keyboardShouldPersistTaps="always"
  style={
    Object {
      "backgroundColor": "#ffffffff",
      "flex": 1,
      "marginLeft": "auto",
      "marginRight": "auto",
      "maxWidth": 650,
      "width": "100%",
    }
  }
>
  <TouchableWithoutFeedback
    onPress={[Function]}
  >
    <View>
      <View
        style={
          Array [
            Object {
              "backgroundColor": "white",
              "flex": 1,
              "justifyContent": "center",
              "width": "100%",
            },
            Object {
              "display": "flex",
              "flexDirection": "row",
            },
          ]
        }
      >
        <FlatList
          columnWrapperStyle={
            Object {
              "flex": 1,
              "justifyContent": "space-between",
            }
          }
          data={Array []}
          disableVirtualization={false}
          extraData={
            Object {
              "activeLink": "Browse Properties",
              "address": "",
              "agentName": "",
              "arrayHolder": Array [],
              "city": "",
              "description": "",
              "epcrating": "",
              "extensiontype": Array [],
              "inputFieldError": Object {
                "City": false,
                "Postcode One": true,
                "Postcode Two": true,
                "Property Description": false,
                "Property Number": true,
                "Property Price": true,
                "Property Title": false,
                "Streen name": false,
                "keyFeature": false,
                "porpstatus": false,
                "propTenure": false,
                "propType": false,
                "selectedBathroom": false,
                "selectedBed": false,
                "selectedImage": false,
              },
              "isBathroomFilled": true,
              "isBoxOpen": false,
              "isCityFilled": true,
              "isDescriptionFilled": true,
              "isImageBoxOpen": false,
              "isKeyFilled": true,
              "isMainImageFilled": true,
              "isNextButtonDisabled": false,
              "isPostcodeOneFilled": true,
              "isPostcodeTwoFilled": true,
              "isPriceModifierFilled": true,
              "isPropNumberFilled": true,
              "isPropPriceFilled": true,
              "isPropTypeFilled": true,
              "isPropertyTitleFilled": true,
              "isSnackbarOpen": false,
              "isStatusFilled": true,
              "isStreetFilled": true,
              "isTenureFilled": true,
              "isaddressfilled": true,
              "isagentfilled": true,
              "isbedFilled": true,
              "isnumberValid": true,
              "keyFeature": Array [],
              "parking": "",
              "phoneNumber": "",
              "porpstatus": "",
              "postcodeOne": "",
              "postcodeTwo": "",
              "propDesInfo": false,
              "propNumber": "",
              "propPrice": "",
              "propTenure": "",
              "propType": "",
              "propertyTitle": "",
              "reception": "",
              "save": false,
              "selectedAgent": "",
              "selectedBathroom": "",
              "selectedBed": "",
              "selectedButton": 0,
              "selectedFloorImage": null,
              "selectedForDeletion": null,
              "selectedImage": null,
              "selectedImageFilename": "",
              "selectedImageSecond": Array [],
              "selectedImageToDelete": Array [],
              "selectedImgeEpc": null,
              "selectedOption": "",
              "selectedPriceModi": "",
              "selectedPropStatus": "Available",
              "streetname": "",
              "token": "",
              "tourlink": "",
              "userLink": "",
            }
          }
          horizontal={false}
          initialNumToRender={10}
          keyExtractor={[Function]}
          maxToRenderPerBatch={10}
          numColumns={2}
          onEndReachedThreshold={2}
          removeClippedSubviews={false}
          renderItem={[Function]}
          scrollEventThrottle={50}
          updateCellsBatchingPeriod={50}
          windowSize={21}
        />
      </View>
    </View>
  </TouchableWithoutFeedback>
</ScrollViewMock>
`;
